### PR TITLE
test-cases to check model-fields and BlogListView

### PR DESCRIPTION
**Pull request by nidhi-tagline**

### CODE REVIEW

File: author/tests.py [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+        
+    def test_bio_detail_label(self):
+        author = self.get_author()
+        field_label = author._meta.get_field('bio_detail').verbose_name
+        self.assertEqual(field_label,'bio detail')
+        
+    def test_object_name(self):
+        author = self.get_author()
+        expected_name = author.username
+        self.assertEqual(expected_name, str(author))
```
Comment (himanshu-suthar-tagline) [inline author/tests.py:32]: Tell me what are you testing here.

File: blog/tests.py [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+        post = comment.post
+        self.assertIsInstance(post,Blog)
+        
+    def test_commentor_object(self):
+        comment = self.get_comment()
+        created_by = comment.created_by
+        self.assertIsInstance(created_by,Author)
+
+
+from django.urls.base import reverse
```
Comment (himanshu-suthar-tagline) [inline blog/tests.py]: Don't import anything in between the file. The modules should be imported at the top only.

File: author/tests.py (hunk 1 above)
Comment (nidhi-tagline) [reply author/tests.py:32]: These are the testcases for author model according to assessment doc. you have provided.

- In first test and third test, I've tested if created author object has correct field label.
- Second test is to check whether max_length of username field is correct.
- Fourth test is to check that  __str__() returns the expected value.

**Review by nidhi-tagline — COMMENTED** [review]
attended changes

File: author/tests.py (hunk 1 above)
Comment (himanshu-suthar-tagline) [reply author/tests.py:32]: Only fourth one makes sense. You should write the test cases to check if all the author details are correct? If blog has the correct author or not. Logged out users can't edit the author, etc like this.

You should not write the test cases to check if column exist or not, they are useless.

**Review by nidhi-tagline — COMMENTED** [review]
I've added few more tests,
- to check author update only accessible to logged in authors
- to check created blog has correct author
- to check that only authenticated user can add comment
- to check if author details are correct
- to check that other user cannot update profile

**Pull request merged**